manifests/init.pp

```diff
@@ -158,6 +179,13 @@
 #
 class consul (
   Hash $acls                                 = $consul::params::acls,
+  Hash $tokens                               = $consul::params::tokens,
+  Hash $policies                             = $consul::params::policies,
+  String $acl_api_hostname                   = $consul::params::acl_api_hostname,
```

Comment:
You can enforce the minimal string length like this:
```suggestion
  String[1] $acl_api_hostname                = $consul::params::acl_api_hostname,
```

Please also look through https://github.com/puppetlabs/puppetlabs-stdlib/tree/master/types, you might find mor strict datatypes that you could use.